src/pages/workspaces/workspace/Data.js

```diff
@@ -369,8 +384,17 @@ const ToolDrawer = _.flow(
   const [toolMode, setToolMode] = useState()
   const entitiesCount = _.size(selectedEntities)
   const isCohort = entityKey === 'cohort'
-  const dataExplorerButtonEnabled = isCohort && entitiesCount === 1 && _.values(selectedEntities)[0].attributes.data_explorer_url
-  const dataExplorerUrl = dataExplorerButtonEnabled ? `${_.values(selectedEntities)[0].attributes.data_explorer_url}&wid=${workspaceId}` : undefined
+
+  const dataExplorerButtonEnabled = isCohort && entitiesCount === 1 && _.values(selectedEntities)[0].attributes.data_explorer_url !== undefined
+  let dataExplorerUrl = dataExplorerButtonEnabled ? _.values(selectedEntities)[0].attributes.data_explorer_url : undefined
+  if (dataExplorerUrl) {
+    dataExplorerUrl = dataExplorerUrl.includes('?') ? `${dataExplorerUrl}&wid=${workspaceId}` : `${dataExplorerUrl}?wid=${workspaceId}`
+  }
+  const openDataExplorerInSameTab = dataExplorerUrl && (dataExplorerUrl.includes('terra.bio') || _.some({ origin: new URL(dataExplorerUrl).origin }, datasets))
+  const dataset = openDataExplorerInSameTab && getDataset(dataExplorerUrl)
+  const dataExplorerPath = openDataExplorerInSameTab && Nav.getLink(dataset.authDomain ?
+    'data-explorer-private' :
+    'data-explorer-public', { dataset: dataset.name }) + '?' + dataExplorerUrl.split('?')[1]
```

Comment:
Here's an alternative: 
```js
const originalDataExplorerUrl = isCohort && entitiesCount === 1 && new URL(_.values(selectedEntities)[0].attributes.data_explorer_url)
const isEmbeddedDE = originalDataExplorerUrl && (dataExplorerUrl.origin.includes('terra.bio') || _.some({ origin: dataExplorerUrl.origin }, datasets))
const [baseDataExplorerUrl, dataExplorerUrlSearch] = originalDataExplorerUrl  ? originalDataExplorerUrl.href.split('?') : []
const augmentedUrl = originalDataExplorerUrl && `${baseDataExplorerUrl}?${qs.stringify({ ...qs.parse(dataExplorerUrlSearch), wid: workspaceId })}`
```

and then below:
```js
h(ModalToolButton, {
  onClick: openDataExplorerInSameTab ? undefined : onDismiss,
  ...(isEmbeddedDE ?
    { href: `#${augmentedUrl.split('#')[1]}` } :
    { href: augmentedUrl, ...Utils.newTabLinkProps })
```

Something like this. Didn't have time to really test, have to run out.

Reply:
I can't use `qs.stringify()` doesn't support fragment params. I considered writing `Utils.mergeFragmentParams()`, but it would be non-trival to cover all the corner cases. Added comment.

This doesn't cover this case where it's a standalone url, but for a dataset in the dataset library.
- `data_explorer_url` is `https://baseline-baseline-explorer.appspot.com/?filter=baseline-baseline-analysis.participant_data_v1.TableA_PhysHealth.sex_at_birth%253DFEMALE&extraFacets=`
- My code figures out this is in dataset library and converts to `http://localhost:3000/#library/datasets/Baseline%20Health%20Study/data-explorer?filter=baseline-baseline-analysis.participant_data_v1.TableA_PhysHealth.sex_at_birth%253DFEMALE&extraFacets=&wid=ac0df5b0-d196-4343-8526-2d779c6117d6`

Reply:
Thanks. I'm still not super comfortable with the let, what do you think about my other suggestions above?

Reply:
Done